src/click/shell_completion.py

```diff
@@ -302,8 +302,10 @@ class BashComplete(ShellComplete):
     def _check_version(self) -> None:
         import subprocess
 
-        output = subprocess.run(["bash", "--version"], stdout=subprocess.PIPE)
-        match = re.search(r"version (\d)\.(\d)\.\d", output.stdout.decode())
+        output = subprocess.run(
+            ["bash", "-c", "echo ${BASH_VERSION}"], stdout=subprocess.PIPE
```

Comment:
Please note that this does not always return the version of `bash` where autocompletion has been invoked from.

Here's an example, taken from one macOS system:
```
$ which -a bash
/usr/local/bin/bash
/bin/bash
$ help | head -1
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)
$ bash -c 'echo ${BASH_VERSION}'
4.4.23(1)-release
$ echo $BASH_VERSION
3.2.57(1)-release
```
Here, /bin/bash is the OS-provided bash 3.2.57, while /usr/local/bin/bash is bash 4.4.23 installed using Homebrew. The user profile is configured with /bin/bash as the login shell.

I don't see a good way to resolve this issue other than to tell the user that such setup is "broken". I'm mentioning it anyway in case someone else has an idea.

Reply:
If `BASH_VERSION` is available in the env, then couldn't we look at that directly?

Reply:
Unfortunately BASH_VERSION is not exported by default, so while
```bash
$ echo ${BASH_VERSION}
4.4.23(1)-release
```
works,
```bash
$ python -c 'import os; print(os.environ.get("BASH_VERSION"))'
None
```
doesn't. This can be worked around by doing `export BASH_VERSION` before invoking `click` for autocompletion, but I think this would be a breaking change.

Reply:
But the current code has the very same problem, as you initially mentioned @igrr, as it also just calls the first bash in `$PATH`.

Reply:
I agree with that and with your proposed solution @janLo, I only mentioned this (extra) issue in case anyone looking at this PR comes up with a way to deal it.